src/librbd/LibrbdWriteback.cc

```diff
@@ -97,6 +104,11 @@ namespace librbd {
   {
     // on completion, take the mutex and then call onfinish.
     Context *req = new C_Request(m_ictx->cct, onfinish, &m_lock);
+    if (!m_ictx->object_may_exist(object_no)) {
+      m_finisher->queue(req, -ENOENT);
```

Comment:
It's also in ```AioRead``` -- it's only here to prevent the cache path from issuing a read request (since ```LibrbdWriteback``` bypasses ```AioRead::send``` to issue the send but uses the remainder of the ```AioRead``` logic for ```::should_complete```).  I have a rainy day idea to have the ```ObjectCacher``` use a generic read request context so that the read logic doesn't need to be duplicated here.